js/src/common/helpers/highlight.tsx

```diff
@@ -1,16 +1,16 @@
+import * as Mithril from 'mithril';
 import { truncate } from '../utils/string';
 
 /**
  * The `highlight` helper searches for a word phrase in a string, and wraps
  * matches with the <mark> tag.
  *
- * @param {String} string The string to highlight.
- * @param {String|RegExp} phrase The word or words to highlight.
- * @param {Integer} [length] The number of characters to truncate the string to.
+ * @param string The string to highlight.
+ * @param phrase The word or words to highlight.
+ * @param [length] The number of characters to truncate the string to.
  *     The string will be truncated surrounding the first match.
- * @return {Object}
  */
-export default function highlight(string, phrase, length) {
+export default function highlight(string: string, phrase: string | RegExp, length: number): Mithril.Vnode | string {
```

Comment:
Length should be `length?: number` as it is an optional parameter.

Reply:
In typescript all parameters are required by default so yes - it should be, fixed.

Reply:
True, but we don't have TS errors or warnings enabled. The TS is just converted into plain JS - type hinting is mainly to help during development 🙂